src/Adapter/MSTest.TestAdapter/Discovery/AssemblyEnumerator.cs

```diff
@@ -301,7 +301,19 @@ private static bool TryProcessTestDataSourceTests(UnitTestElement test, TestMeth
     {
         foreach (var dataSource in testDataSources)
         {
-            var data = dataSource.GetData(methodInfo);
+            IEnumerable<object?[]>? data = null;
+            try
+            {
+                data = dataSource.GetData(methodInfo);
+            }
+            catch (Exception ex) when (ex is ArgumentException && MSTestSettings.CurrentSettings.ConsiderEmptyDataSourceAsInconclusive)
+            {
+                var discoveredTest = test.Clone();
+                discoveredTest.DisplayName = dataSource.GetDisplayName(methodInfo, null) ?? discoveredTest.DisplayName;
+                tests.Add(discoveredTest);
```

Comment:
We probably don't need the clone here as we won't create multiple instances.

Reply:
Actually maybe we need it for the other data source. Can you please do a test where you would have 2 dynamic data, one empty and one non-empty to see if that's needed?